src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs

```diff
@@ -567,6 +567,24 @@ public T[] ToArray()
         }
 
         /// <summary>
+        /// Removes all values from the <see cref="ConcurrentBag{T}"/>.
+        /// </summary>
+        public void Clear()
+        {
+            // Short path if the bag is empty
+            if (_headList == null)
+                return;
+
+            // Acquire the global lock to update the pointers
+            lock (GlobalListsLock)
+            {
+                _locals = new ThreadLocal<ThreadLocalList>();
+                _headList = null;
+                _tailList = null;
```

Comment:
This is not a safe change.  It would need to synchronize with all of the lists, which means freezing the bag before making these changes.  And you shouldn't need to allocate a new thread local; you should be able to freeze the bag, iterate through all of the local lists clearing each, then unfreeze the bag.